projects/plugins/jetpack/modules/masterbar/admin-menu/class-jetpack-admin-menu.php

```diff
@@ -13,6 +13,41 @@
  * Class Jetpack_Admin_Menu.
  */
 class Jetpack_Admin_Menu extends Admin_Menu {
+
+	/**
+	 * Returns an array with the CPT menu items different form post and page types.
+	 *
+	 * @param array $menu The menu array to filter from.
+	 * @return array The filtered menu array.
+	 */
+	private function filter_cpt_menu_items( $menu ) {
+		$exclude_slugs = array( 'menu-posts', 'menu-pages' );
+		return array_filter(
+			$menu,
+			function ( $item ) use ( $exclude_slugs ) {
+				$is_edit_post_type = strpos( $item[2], 'edit.php?post_type=' ) === 0;
+				return $is_edit_post_type && ! in_array( $item[5], $exclude_slugs, true );
+			}
+		);
+	}
+
+	/**
+	 * Replaces the wp-admin CPT URLs with the WPcom calypso url.
+	 *
+	 * @param array $menu_item The CPT menu item to replace.
+	 * @return array The CPT menu item with the replaced URL.
+	 */
+	private function replace_admin_link_to_calypso( $menu_item ) {
```

Comment:
One can easily infer with this method name that all WP-Admin links will be replaced, while only the CPT ones are replaced.

For consistency with how the rest of the menus are added/modified, I recommend creating a new method like `add_cpt_menus` and keep the `$menu = array()` reset.

This new method would need to iterate over [all the registered CPTs](https://developer.wordpress.org/reference/functions/get_post_types/) (remember to use the `_builtin` param to exclude the default post types!) and register the new menu items ([if `show_in_menu` is enabled](https://developer.wordpress.org/reference/functions/register_post_type/#show_in_menu)) with a link to Calypso (or to WP Admin is the SSO module is disabled) and ensure they are positioned below the Feedback menu item (we could also register the Feedback menu item in this method and get rid of `add_feedback_menu`).

Reply:
@mmtr , Thanks for your suggestions. I like this new approach.

The PR is ready to review, and I've updated the description and test instructions to match the new behavior.

My only question is should we show only the CPTs with `show_in_rest` equal to true?.
In the example provided in the issue https://github.com/Automattic/wp-calypso/issues/64935 we had this option `show_in_rest` equal to true. However, I think we should show all the CPT available on the menu, regardless of the `show_in_rest` value.
In this commit, I've removed filtering by this option: https://github.com/Automattic/jetpack/pull/25670/commits/ef4a1764291578519a5108fdfe1a62566c1b570a

Reply:
>  should we show only the CPTs with show_in_rest equal to true?

I think we should ignore that, since it could led to a confusing behavior (why a CPT that is registered to show up in the REST API but not in the admin menu should appear in the Calypso menu?